Include/internal/pycore_interp.h

```diff
@@ -230,6 +238,28 @@ PyAPI_FUNC(int) _PyInterpreterState_IDInitref(PyInterpreterState *);
 PyAPI_FUNC(int) _PyInterpreterState_IDIncref(PyInterpreterState *);
 PyAPI_FUNC(void) _PyInterpreterState_IDDecref(PyInterpreterState *);
 
+
+#if WITH_FREELISTS
+static inline PyObject*
+_PyInterpreterState_FreelistAlloc(PyInterpreterState *interp, int size) {
+    assert(size >= 4);
+    assert((size & 0x1) == 0);
+    int index = (size-4)/2;
+    return _PyFreeList_Alloc(&interp->freelists[index]);
+}
+
+static inline void
+_PyInterpreterState_FreelistFree(PyInterpreterState * interp, PyObject *op, int size) {
```

Comment:
What is `size`?. The assumption is that it is size in bytes, but I think you are using size in machine words.
Also needs to be `Py_ssize_t`.